### PR TITLE
Enable fake hot/cold splitting on ARM64

### PR DESCRIPTION
**Pull request by amanasifkhalid**
This PR enables fake hot/cold splitting on ARM64 via `COMPlus_JitFakeProcedureSplitting`, and fixes various bugs exposed by testing with fake-splitting:

- Branches between hot/cold sections are now always long.
- The pseudoinstruction for loading a constant from the cold section did not support loading 16-byte data into vector registers, as it temporarily loaded the constant into an 8-byte integer register. Now, 16-byte constants are loaded directly into vector registers via an `ld1` instruction.
- Tests involving loading 16-byte constants exposed the data section is not always aligned to its largest constant. Now, the data section is always aligned to `emitConsDsc.alignment` when calling `eeAllocMem`.
- Asserts/NYIs blocking hot/cold splitting on ARM64 have been removed.

Fake hot/cold splitting requires we fake unwind info by treating each split function as one hot section. A more architecture-agnostic approach for this has been applied. However, unwind info generated for cold sections has yet to be tested, as this depends on VM support for hot/cold splitting.

### CODE REVIEW

**Comment by amanasifkhalid** [comment]
Closing to prevent re-running checks.

**Comment by amanasifkhalid** [comment]
I realized my implementation for aligning the data section when fake-splitting is overzealous, and simply tweaking the current implementation to align to `TARGET_POINTER_SIZE` when `emitConsDsc.alignment >= TARGET_POINTER_SIZE` (rather than simply `==`) fixes the asserts hit when loading 16-byte constants.

@BruceForstall PTAL. I noticed `runtime-jit-experimental` is failing due to some odd concurrency issue (at least that's what it looks like). Hopefully this was just a fluke.

**Comment by BruceForstall** [comment]
> I noticed `runtime-jit-experimental` is failing due to some odd concurrency issue (at least that's what it looks like). Hopefully this was just a fluke.

Looks like a (bad) recent regression. I opened https://github.com/dotnet/runtime/issues/71023.

**Comment by amanasifkhalid** [comment]
@BruceForstall PTAL at `eeAllocMem`. I incorporated your suggestions to simplify the caller logic and improve the method's abstraction: `eeAllocMem` now takes `emitConsDsc.alignment` as an additional argument since it's already calculated, and handles the ARM64-specific fix-ups. The caller no longer handles anything ARM64- or fake-splitting-specific.

**Comment by BruceForstall** [comment]
> @BruceForstall PTAL at `eeAllocMem`. I incorporated your suggestions to simplify the caller logic and improve the method's abstraction: `eeAllocMem` now takes `emitConsDsc.alignment` as an additional argument since it's already calculated, and handles the ARM64-specific fix-ups. The caller no longer handles anything ARM64- or fake-splitting-specific.

I like this direction. Unfortunately, and I apologize, this change is going to clash with my change https://github.com/dotnet/runtime/pull/71044. Can we wait until that is merged (today), and then review this? You'll need to pick up its changes into this.

**Comment by amanasifkhalid** [comment]
> I like this direction. Unfortunately, and I apologize, this change is going to clash with my change https://github.com/dotnet/runtime/pull/71044. Can we wait until that is merged (today), and then review this? You'll need to pick up its changes into this.

No worries!

**Comment by amanasifkhalid** [comment]
@BruceForstall I merged your changes from #71044 in, and added a guard to disable splitting on LoongArch64. Thanks for the ARM64 alignment work -- I recall hitting some related asserts when trying to load 16-byte constants a few weeks ago...

**Review by BruceForstall — APPROVED** [review]
LGTM

File: src/coreclr/jit/unwind.cpp
Comment (amanasifkhalid) [reply src/coreclr/jit/unwind.cpp:277]: If I recall correctly, not setting `pColdCode = nullptr` here results in two calls to `eeAllocUnwindInfo`, breaking our implementation for fake unwind info. I agree nulling `pColdCode` so late doesn't look great; my unwind info work is largely patchwork to minimize code churn, but I can look into re-architecting this.

**Pull request merged**
